FWCore/Framework/src/EventPrincipal.cc

```diff
@@ -323,12 +325,15 @@ namespace edm {
         continue;
       }
       // Get the thinned container and return a pointer if we can find it
```

Comment:
@Dr15Jones @makortel per the discussion today, one alternative to checking ProductResolver::unscheduledWasNotRun() in getThinnedAssocation would be to move the check here, befre even calling getThinnedAssocation, by retrieving the product id directly fromt he associationsHelper using branchIDToProductID (I didn't go for this option originally because it looks like that's a relatively slow lookup operation)